backend/libraries/ballerina-core-python/ballerina_core/parsing/parsing_types.py

```diff
@@ -1,13 +1,34 @@
 from __future__ import annotations
 
 from collections.abc import Callable, Mapping, Sequence
+from dataclasses import dataclass
 from decimal import Decimal
 from typing import TypeVar
 
+from ballerina_core.sum import Sum
+
 Json = Mapping[str, "Json"] | Sequence["Json"] | str | int | Decimal | bool | None
 
+
+@dataclass(frozen=True)
+class ParsingError:
```

Comment:
would call this `ParsingErrorContext`

Reply:
It is the error (that also contains context)